support/xspec-dev/readme.md

```diff
@@ -0,0 +1,96 @@
+
+The intention being to merge all this in to ../xspec at some point tbd.
+
+The initiative focuses on two kinds of assets:
+
+- XProc pipelines to orchestrate XSpec evaluation
+- Scripts to initiate XProc pipelines
+
+These should be either useful directly, or useful if copied and adjusted, i.e. as templates.
+
+Additionally, this project has new functionality for XSpec provided by an updated **XSpec Report HTML** transformation.
+
+## Pipelines
+
+The pipelines in this project depend on prior art in the XSpec repository. Pipelines are currently implemented using XProc 1.0 to be run under XML Calabash.
+
+XProc 3.0 is planned for a later date.
+
+Three pipelines are provided at the top level. All have dependencies on XSpec [in the XSpec distribution included as a repo submodule](../xspec/src/harnesses/). Rather than forking from XSpec, the aim is to merge these capabilities back in.
+
+`xspec-single-report.xpl` - processes an XSPec input file by applying XSpec execution using the XSpec XProc test harness, then processing its native report to make an HTML page.
+
+The HTML XSLT was developed to mitigate configuration problems using the project-standard presentation stylesheet (as noted below). It is designed to be functional but lightweight, robust and cleanly extensible for a range of use cases or deployments.
+
+*Limitation*: At present, the presentation stylesheet meets these goals with respect to XSpec testing XSLT - not (yet) testing XQuery, Schematron or other runtimes to which XSpec might be applied.
+
+`xspec-batch-report.xpl` - provides the same set of operations, except instead of a single XSpec (file instance), it can accept any number of XSpec files designated as inputs (either wired in or via XProc port bindings). Results from these tests are *aggregated* into a single report.
+
+This pipeline also demonstrates how reports can be further processed to provide other outputs such as XML summaries (for batches) and simple plain-text summaries.
+
+Additionally, another XProc is included for future diagnostics:
+
+`xspec-single-xspec-repo-report.xpl` - calls the XSpec project XProc (pipeline), delivering an HTML file using the 'regular' XSLT in the XSpec distribution. *Note*: YMMV especially as it regards CSS setup in HTML file results.
+
+This pipeline is essentially a minimal, functionally standalone wrapper around XSpec-native capabilities, made to expose the 'pain points' in using the current resources (which were originally authored and tested over ten years ago). For good results using XProc 1.0 under XML Calabash (itself no longer cutting-edge technology), certain modifications must be made in the XSpec repository libraries:
+
+* Serialization needs to use latest (HTML5) serialization options to deal with character handling, rather than the current 'double reverse' character escaping of markup characters into the Unicode PUA - thus effectively requiring a character mapping back, which is not well supported under XProc 1.0.
```

Comment:
Non-blocking: it may help the uninitiated like me:

```suggestion
* Serialization needs to use latest (HTML5) serialization options to deal with character handling, rather than the current 'double reverse' character escaping of markup characters into [the Unicode Private Use Areas (PUA)](https://en.wikipedia.org/wiki/Private_Use_Areas) - thus effectively requiring a character mapping back, which is not well supported under XProc 1.0.
```

Reply:
done in latest commit